### ISSUE TITLE
Nested overlays incompatible with screenreader

### ISSUE THREAD

**Issue by GuidoThomassen** [ing-bank/lion#1431]
When using nested overlays, it is possible with a screenreader to go to the elements behind the overlay by swiping left.

## Expected behavior

Swiping left with a screenreader goes to the last element within the same overlay.

## Actual Behavior

Swiping left with a screenreader goes to the element at the bottom of the page outside of the overlay.
In the preview I am using the example from the Lion docs: https://lion-web.netlify.app/docs/systems/overlays/configuration/#isblocking

![Lion overlay bug](https://user-images.githubusercontent.com/16132138/123088472-0e60c200-d426-11eb-87f9-9a809768f28d.gif)

**Comment by gerjanvangeest** [comment]
The example you used does not have the `trapsKeyboardFocus` option enabled, does the same issue occur in this demo:
https://lion-web.netlify.app/docs/systems/overlays/configuration/#trapskeyboardfocus ?

**Comment by GuidoThomassen** [comment]
@gerjanvangeest No it does not occur on that demo, but that one doesn't contain nested overlays right?

**Comment by gerjanvangeest** [comment]
Yes, but you can combine any config item to create your own config:

```js
  const isBlockingAndKeepsFocusConfig = { hasBackdrop: true, isBlocking: true, trapsKeyboardFocus: true };
```

Our default dialog config can be found here: https://github.com/ing-bank/lion/blob/master/packages/overlays/src/configurations/withModalDialogConfig.js

I'll close the issue, since the behaviour works as expected. If you still have questions, just post them below.

**Comment by GuidoThomassen** [comment]
Hi @gerjanvangeest,

Yes you are right that on the original reproduction the trapsKeyboardFocus was not included.

However, 
I looked into this again and have to conclude that there is unfortunately a bug. In our app we are using LionDialog which already has these options enabled (withModalDialogConfig). We've created a repo where you can see the bug reproduced: https://github.com/GuidoThomassen/lion-dialog-bug

![Reproduction](https://user-images.githubusercontent.com/16132138/123103065-f2fdb300-d435-11eb-95c4-439bf1489261.gif)

**Comment by gerjanvangeest** [comment]
That looks indeed like a bug, thanks for double checking :)

If you have the time, a contribution with a fix would be highly appreciated.
